modules/videojsVideoProvider.js

```diff
@@ -0,0 +1,131 @@
+  RENDITION_UPDATE, ERROR, COMPLETE, PLAYLIST_COMPLETE, FULLSCREEN, PLAYER_RESIZE, VIEWABLE, CAST, PLAYBACK_MODE
+} from './videoModule/constants/events.js';
+import stateFactory from './videoModule/shared/state.js';
+import { adStateFactory, timeStateFactory,  callbackStorageFactory} from './jwplayerVideoProvider.js';
+
+export function VideojsProvider(config, videojs_, adState_, timeState_, callbackStorage_, utils) {
+  let videojs = videojs_;
+  let player = null;
+  let playerVersion = null;
+  const {playerConfig, divId} = config;
+
+  let adState = adState_;
+  let timeState = timeState_;
+  let callbackStorage = callbackStorage_;
+  let minimumSupportedPlayerVersion = 'v7.17.0';
+
+
+  function init() {
+    console.log("Initialized videojs provider")
+
+    if (!videojs) {
+      triggerSetupFailure(-1); // TODO: come up with code for player absent
+      return;
+    }
+
+    // playerVersion = videojs.version;
+
+    // if (playerVersion < minimumSupportedPlayerVersion) {
+    //   triggerSetupFailure(-2); // TODO: come up with code for version not supported
+    //   return;
+    // }
+
+    // player = videojs(divId);
+    // if (player.getState() === undefined) {
+    //   setupPlayer(playerConfig);
+    // } else {
+    //   setupCompleteCallback && setupCompleteCallback(SETUP_COMPLETE, getSetupCompletePayload());
+    // }
+  }
+
+  function getId() {
+    return divId;
+  }
+
+  function getOrtbParams() {
+    console.log("Requested ortb params")
+    // if (!player) {
+    //   return;
+    // }
+    // const config = player.getConfig();
+    // const adConfig = config.advertising || {};
+    supportedMediaTypes = supportedMediaTypes || utils.getSupportedMediaTypes(MEDIA_TYPES);
+
+    const video = {
+      mimes: [],
+      w: 0, 
+      h: 0,
+    };
+
+    const content = {
+      id: item.mediaid,
+      url: item.file,
+      title: item.title,
+      cat: item.iabCategories,
+      keywords: item.tags,
+      len: duration,
+    };
+
+    return {
+      video,
+      content
+    }
+  }
+
+  function setAdTagUrl(adTagUrl, options) {
+    console.log("Set ad tag url:", adTagUrl)
+    // if (!player) {
+    //   return;
+    // }
+    // player.playAd(adTagUrl || options.adXml, options);
+  }
+
+  function onEvents(events, callback) {
+    console.log("Added callback for", events)
+  }
+
+  function offEvents(events, callback) {
+    console.log("Removed callback for", events)
+
+  }
+
+  function destroy() {
+    console.log("Destroying player")
+    if (!player) {
+      return;
+    }
+    player.remove();
+    player = null;
+  }
+
+  return {
+    init,
+    getId,
+    getOrtbParams,
+    setAdTagUrl,
+    onEvents,
+    offEvents,
+    destroy
+  };
+}
+
+import { VIDEO_JS_VENDOR } from './videoModule/constants/vendorCodes.js';
+import { videoVendorDirectory } from './videoModule/vendorDirectory.js';
```

Comment:
Import statements always go on top in JS. I don't think it affects functionality in any way, but it's the industry norm.